msal/src/Microsoft.Identity.Client/ClaimsPrincipalExtension.cs

```diff
@@ -0,0 +1,108 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Security.Claims;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Extension methods for ClaimsPrincipal
```

Comment:
>/ Extension methods for ClaimsPrincipal [](start = 6, length = 39)

While I understand the benefit it can bring, what are the scenarios? For web only? Also what is the next step?

Reply:
It's for web, when ASP.NET sets a ClaimPrincipal for the signed-in user, available everywhere in the application. If customers want to get a token for an account, it's a confort method to get it from the claims principal.
not sure for DisplayName, BTW.

Reply:
@jmprieur, @henrik-me  - let's decide on whether we want both, one or none of these methods before cleaning up / adding unit tests.

Reply:
yes. Let's decide first. I'm starting to think that we would rather have another library:
Microsoft.Identity.Client.AspNetWeb or something like that.